src/admin-events/admin-events.js

```diff
@@ -0,0 +1,96 @@
+const { Finding, FindingSeverity, FindingType } = require('forta-agent');
+const { getAbi, getAddress } = require('@uma/contracts-node');
+const ethers = require('ethers');
+
+// load config files
+const { umaEverestId } = require('../../agent-config.json');
+const adminEvents = require('./admin-events.json');
+
+// returns the list of events for a given contract
+function getEvents(contractName) {
+  const events = adminEvents[contractName];
+  if (events === undefined) {
+    return []; // no events for this contract
+  }
+  return events;
+}
+
+// get contract names for mapping to events
+const contractNames = Object.keys(adminEvents);
+
+// Create the interfaces for each contract that has events we wish to monitor
+const ifaces = {};
+contractNames.forEach((contractName) => {
+  // Get the abi for the contract
+  const abi = getAbi(contractName);
+
+  // create ethers interface object
+  const iface = new ethers.utils.Interface(abi);
+
+  // Create an association between the contract name and the interface
+  ifaces[contractName] = iface;
+});
+
+// Filters the logs to only events in eventNames
+function filterAndParseLogs(logs, address, iface, eventNames) {
+  // collect logs only from the contract
+  const contractLogs = logs.filter((log) => log.address === address);
+
+  // decode logs and filter on the ones we are interested in
+  const parse = (log) => iface.parseLog(log);
+  const filter = (log) => eventNames.indexOf(log.name) !== -1;
+  const parsedLogs = contractLogs.map(parse).filter(filter);
+
+  return parsedLogs;
+}
+
+// helper function to create alerts
+function createAlert(eventName, contractName, contractAddress, eventType, eventSeverity) {
+  return Finding.fromObject({
+    name: 'UMA Admin Event',
+    description: `The ${eventName} event was emitted by the ${contractName} contract`,
+    alertId: 'AE-UMA-ADMIN-EVENT',
+    type: FindingType[eventType],
+    severity: FindingSeverity[eventSeverity],
+    everestId: umaEverestId,
+    protocol: 'uma',
+    metadata: {
+      contractName,
+      contractAddress,
+      eventName,
+    },
+  });
+}
+
+async function handleTransaction(txEvent) {
+  const findings = [];
+  const chainId = 1;
```

Comment:
Move the chainId constant out of the function?
Also should this be CHAIN_ID according to the Google JS style?

Reply:
Done